### PR TITLE
Problem: Missing memo filter on Account APIs

### LINKED ISSUE #455: Problem: Missing memo filter on Account APIs
Some of the exchanges have been using memo to distinguish user deposit. To facilitate those users tracking their transactions, we can introduce a memo filter to the account APIs

- Account API
- Account messages API
- and other projects related

### PR DESCRIPTION
**Pull request by ysong42**
Solution: fixed #455 by introducing the following changes:

### API Changes:
- Added filter on `/accounts/{account}/transactions?memo={memo}` api
- Support `account/memo` as search `keyword` on `/search?keyword={keyword}` api

### Table Changes:
Introduce new rows to `view_account_transactions_total` table. 

Previously, `identity` column's value could be in `{account}:-` or `{account}:{messageType}` format. 

Now there are two new format of identities, `{account}/{memo}:-` and `{account}/{memo}:{messageType}`, whose `{memo}` are always non-empty string.

### Screenshots

Example: (list account/memo transactions)
`http://localhost:8080/api/v1/accounts/cro1w2kvwrzp23aq54n3amwav4yy4a9ahq2kz2wtmj/transactions?memo=3910079064`

image

Example: (search by account/memo)
`http://localhost:8080/api/v1/search?keyword=cro1w2kvwrzp23aq54n3amwav4yy4a9ahq2kz2wtmj/3910079064`

image

Example: (list account transactions)
`http://localhost:8080/api/v1/accounts/cro1w2kvwrzp23aq54n3amwav4yy4a9ahq2kz2wtmj/transactions`
image

### CODE REVIEW

File: projection/account_transaction/view/account_transactions.go [hunk 1]
```diff
@@ -130,7 +138,12 @@ func (accountMessagesView *AccountTransactions) List(
 	).WithCustomTotalQueryFn(
 		func(rdbHandle *rdb.Handle, _ sq.SelectBuilder) (int64, error) {
 			totalView := NewAccountTransactionsTotal(rdbHandle)
+
 			identity := fmt.Sprintf("%s:-", filter.Account)
+			if filter.Memo != "" {
+				identity = fmt.Sprintf("%s/%s:-", filter.Account, filter.Memo)
+			}
```
Comment (davcrypto) [inline projection/account_transaction/view/account_transactions.go:145]:
```suggestion
			} else {
				identity := fmt.Sprintf("%s:-", filter.Account)
			}
```

Comment (davcrypto) [inline projection/account_transaction/view/account_transactions.go:145]:
```suggestion
			} else {
				identity := fmt.Sprintf("%s:-", filter.Account)
			}
```

**Review by davcrypto — CHANGES_REQUESTED** [review]
May be we can be consistent with other memo checker

**Comment by davcrypto** [comment]
Only a minor comment. Other than that LGTM.

**Comment by ysong42** [comment]
> Only a minor comment. Other than that LGTM.

Thanks for the review! The PR is updated~

**Review by davcrypto — APPROVED** [review]

**Pull request merged**